aptos-move/framework/src/extended_checks.rs

```diff
@@ -428,6 +435,87 @@ impl<'a> ExtendedChecker<'a> {
     }
 }
 
+// ----------------------------------------------------------------------------------
+// Events
+
+impl<'a> ExtendedChecker<'a> {
+    fn check_and_record_events(&mut self, module: &ModuleEnv) {
+        for ref struct_ in module.get_structs() {
+            if self.has_attribute_iter(struct_.get_attributes().iter(), EVENT_STRUCT_ATTRIBUTE) {
+                let module_id = self.get_runtime_module_id(module);
+                // Remember the runtime info that this is a event struct.
+                self.output
+                    .entry(module_id)
+                    .or_default()
+                    .struct_attributes
+                    .entry(
+                        self.env
+                            .symbol_pool()
+                            .string(struct_.get_name())
+                            .to_string(),
+                    )
+                    .or_default()
+                    .push(KnownAttribute::event());
+            }
+        }
+        for fun in module.get_functions() {
+            if fun.is_inline() || fun.is_native() {
```

Comment:
why don't we check inline functions? is that because it is already expressed in the full form of an actual function?

Reply:
Leave a comment please

Reply:
copied from @wrwg 's PR. I think it is `0x1::event::emit` is not an inline.

Reply:
confirmed with him that you're right. Here are just declarations and the actually function called this inline function would have the actual check.